### PR TITLE
Render route in replay history example

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

The ReplayHistoryExample is using the setRoute event, but none of us know it because we're not rendering it. This adds the route line views to the example so we can see it.

### Screenshots or Gifs

This video is also showing that, when you use ReplayHistory it will replay the reroutes. Note though, that the reroute is competing with history files and real-time reroutes. We may want to add the [ability to disable re-routes](https://github.com/mapbox/mapbox-navigation-android/issues/3455)

https://user-images.githubusercontent.com/3021882/126366337-57b3d418-6ad6-4aba-9ff3-28c0280d79f6.mp4

### CODE REVIEW

**Comment by kmadsen** [comment]
this is on top of this one https://github.com/mapbox/mapbox-navigation-android/pull/4636

pull requests are separate to show how the issues are different

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt [hunk 1]
```diff
@@ -165,6 +181,61 @@ class ReplayHistoryActivity : AppCompatActivity() {
...
+        MapboxRouteLineOptions.Builder(this)
+            .withRouteLineResources(
+                RouteLineResources.Builder()
+                    .routeLineColorResources(
+                        RouteLineColorResources.Builder().build()
+                    )
+                    .build()
+            )
+            .withRouteLineBelowLayerId("road-label")
+            .withVanishingRouteLineEnabled(true)
```
Comment (cafesilencio) [inline examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:196]: The video in the PR doesn't show the route line vanishing. The example doesn't have all of the necessary configuration for it to work properly. I suggest setting this to false if you don't intend to use it in the example.

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt [hunk 2]
```diff
@@ -165,6 +181,61 @@ class ReplayHistoryActivity : AppCompatActivity() {
...
+            routeLines
+        ) { value ->
+            binding.mapView.getMapboxMap().getStyle()?.apply {
+                routeLineView.renderRouteDrawData(this, value)
+            }
+        }
+    }
+
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        routeLineApi.updateWithRouteProgress(routeProgress)
```
Comment (cafesilencio) [inline examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:232]: You don't need this if your NOT using the vanishing route line feature.

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt (hunk 1 above)
Comment (kmadsen) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:196]: Hmm I think this is because it's not map matching 🤔

Comment (kmadsen) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:196]: All our examples are broken with the map matcher. Check our the Routeline and Route Arrow Example

Screen Shot 2021-07-19 at 3 26 23 PM

Comment (kmadsen) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:196]: https://github.com/mapbox/mapbox-navigation-android/issues/4640

Comment (cafesilencio) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:196]: If you meant to use the vanishing route line in your example you're missing the `addOnIndicatorPositionChangedListener`. There maybe some other issue if as you say the other examples aren't working but your example specifically is missing that listener.

Comment (cafesilencio) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:196]: The route line and arrow example is working (vanishing) for me on the main branch.

Comment (kmadsen) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:201]: Updated!

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt (hunk 2 above)
Comment (kmadsen) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:237]: I did want to have vanishing route line in this, so let's make sure it's wired correctly. Looks like this was a missing piece https://github.com/mapbox/mapbox-navigation-android/pull/4637#discussion_r672674233

Comment (Guardiola31337) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt:237]: Yes please! I want to replay some routes and double check if I can reproduce some vanishing issues I run into 🚀

**Review by cafesilencio — APPROVED** [review]

**Pull request merged**